### DISCUSSION TITLE
캐릭터 이동을 못하게

### DISCUSSION THREAD

**Discussion by hjs6082** [naverz/zepeto-studio-kor#1218] [5. Q&A]
전시회 기능 개발중입니다. 패널이뜨면 캐릭터 이동을 못하게하려하는데
ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.additionalRunSpeed -= 5;
                ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.additionalJumpPower -= 13;
                ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.additionalWalkSpeed -= 2;
이건 적용이 되는데 애니메이션이보이고
ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.additionalRunSpeed = 0;
                ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.additionalJumpPower = 0;
                ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.additionalWalkSpeed = 0
은 아예 작동되지않습니다.

캐릭터 이동을 못하게 캐릭터 이동 스크립트를 enable 한다던지 하는 방법이 있을까요? 방법 알려주세요

**Comment by zlmrzlo** [comment]
-=로 하셨으면 반대로 하고 싶으면 +=로 해보시면 어떨까요?

**Reply by hjs6082** [reply]
그게 무슨소리입니까.. 이동을 금지하고싶은데 +를하면 속도가빨라져요..

**Reply by zlmrzlo** [reply]
아래 koldshine님의 1번 추천드립니다.

**Comment by koldshine** [comment]
단순하게 이동만 못하게 하실려면
 
1. ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.enable = false  
2. 혹은 말씀하신 패널의 canvas가 screen space이면 blcoking mask로 모든 터치를 막는 방법도 있습니다.
3. 이동스틱UI에 접근이 가능하면 잠시 안 보이게 하는 방법도 있겠네요

상황에 맞게 쓰시면 될 듯 하네요

**Comment by YSHFM** [comment]
this.gameObject.transform.position = new Vector3(x,y,z);
Physics.SyncTransforms();
사용하면 해당 좌표로 이동됩니다. 특정조건에 해당 좌표로 고정시키고 있으면 되겠네요.
